From 03474e1e24b7c3e4519a2be0af5eff5e59973b4a Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 19 Sep 2017 18:56:38 +0800
Subject: [PATCH 2576/5242] MLK-16536-2 video: fbdev: dcss: refine 'esize'
 obtainment

commit  d687f3635071c9789681fa2878969dadd3da4b02 from
https://source.codeaurora.org/external/imx/linux-imx.git

The 'esize' is better be got by 'kfifo_esize()'
instead of using 'sizeof' to make the code more
compatible and easier to unserstand.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index e085942..37e0e0c 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2233,7 +2233,7 @@ static void dcss_ctxld_config(struct work_struct *work)
 	chans = &info->chans;
 	cfifo = &info->cfifo;
 	dsb_len = cc->sb_data_len + cc->db_data_len;
-	esize = sizeof(struct ctxld_unit);
+	esize = kfifo_esize(&cfifo->fifo);
 
 	sg_init_table(cfifo->sgl, cfifo->sgl_num);
 	nsgl = kfifo_dma_out_prepare(&cfifo->fifo, cfifo->sgl,
-- 
1.7.9.5

